### PR TITLE
HHH-17188 Query Cache contains null values when entity is found in Persistence Context

### PR DESCRIPTION
**Pull request by mbladel**
https://hibernate.atlassian.net/browse/HHH-17188

### CODE REVIEW

**Comment by mbladel** [comment]
Thanks @beikov for the suggested solution leveraging the existing assembler logic, updated the PR.

File: hibernate-core/src/main/java/org/hibernate/sql/results/graph/DomainResultAssembler.java
Comment (beikov) [inline hibernate-core/src/main/java/org/hibernate/sql/results/graph/DomainResultAssembler.java:1]: Another optimization you could do is to implement the `resolveState` method in `BasicResultAssembler` to just call `extractRawValue` and `org.hibernate.metamodel.mapping.internal.CompoundNaturalIdMapping.AssemblerImpl` to recursively call `resolveState` on the nested assemblers.

**Review by beikov — APPROVED** [review]
Other than what I commented, this looks good. I was trying hard to come up with a possible reason for moving the `resolveState()` call into the `initializeInstance` phase but I was unable to come up with a scenario that could break, so I guess this is fine for now :)

File: hibernate-core/src/main/java/org/hibernate/sql/results/graph/DomainResultAssembler.java
Comment (mbladel) [reply hibernate-core/src/main/java/org/hibernate/sql/results/graph/DomainResultAssembler.java:1]: Applied the suggestion, thanks.

**Review by beikov — APPROVED** [review]

**Pull request merged**